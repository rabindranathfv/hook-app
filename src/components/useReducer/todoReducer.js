const ADD = 'add';
const DEL ='delete';
const TOG = 'toggle';
const TOG_OLD = 'toggle-old';

export const todoReducer = ( state = [], action ) => {

    switch ( action.type ) {
        case ADD:
            return [ ...state, action.payload ];
        case DEL:
            return state.filter( todo => todo.id !== action.payload );
        case TOG: 
            return state.map( todo => 
                ( todo.id === action.payload )
                    ? { ...todo, done: !todo.done }
                    : todo
            );
        case TOG_OLD:
            return state.map( todo => {

                if ( todo.id === action.payload ) {
                    return {
                        ...todo,
                        done: !todo.done
                    }
                } else {
                    return todo;
                }
            })
        default:
            return state;
    }


}