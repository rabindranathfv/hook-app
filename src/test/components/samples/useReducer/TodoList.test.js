import React from 'react';
import { shallow } from 'enzyme';
import { TodoList } from '../../../../components/useReducer/TodoList';

const todosMock = [
    { id: 1, desc: 't1', done: false},
    { id: 2, desc: 't2', done: false},
    { id: 3, desc: 't3', done: true }
]

const handleDeleteMock = jest.fn();
const handleToggleMock = jest.fn();
describe('TodoList', () => {

    let wrapper = shallow(<todoList todos={todosMock} />);

    beforeEach(() => {
        wrapper = shallow(<TodoList todos={todosMock} />);
    });

  it('should render todoList', () => {
    expect(wrapper).toMatchSnapshot();      
  });

  it('should render all todos', () => {
      const todoList = wrapper.find('ul');
      
      expect(todoList.props().children.length).toBe(3);
  });
});
