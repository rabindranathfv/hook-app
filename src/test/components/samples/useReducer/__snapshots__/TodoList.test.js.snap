// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render todoList 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    index={0}
    key="1"
    todo={
      Object {
        "desc": "t1",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    index={1}
    key="2"
    todo={
      Object {
        "desc": "t2",
        "done": false,
        "id": 2,
      }
    }
  />
  <TodoListItem
    index={2}
    key="3"
    todo={
      Object {
        "desc": "t3",
        "done": true,
        "id": 3,
      }
    }
  />
</ul>
`;
